core/src/main/java/org/apache/accumulo/core/clientImpl/ClientContext.java

```diff
@@ -729,14 +787,18 @@ public AuthenticationToken token() {
 
   @Override
   public void close() {
-    closed = true;
+    closed.compareAndSet(false, true);
     if (thriftTransportPool != null) {
       thriftTransportPool.shutdown();
     }
     if (tableZooHelper != null) {
       tableZooHelper.close();
     }
     singletonReservation.close();
+    this.scannerReadaheadPool.shutdownNow(); // abort all tasks, client is shutting down
+    this.scannerPoolCleaner.clean();
+    this.cleanupThreadPool.shutdown(); // wait for shutdown tasks to execute
+    this.cleaner.clean();
```

Comment:
I think these cleaners can go away, based on my previous comments.

Reply:
Addressed in 180654d